### PR TITLE
Move buckets into kettle/buckets.yaml & validate their prefixes.

### PR DESCRIPTION
**Pull request by rmmh**
Also, make Kettle automatically use the latest buckets.yaml-- this means
nobody needs to spin up a new kettle image when buckets.yaml changes.

### CODE REVIEW

File: docs/tests/node-e2e-ci.md [hunk 1]
```diff
@@ -158,7 +158,7 @@ Official description of the individual files and their content is described by [
 ## Publishing test results in the TestGrid
 
 To have the [TestGrid](https://k8s-testgrid.appspot.com/) consume the new build results, one needs to extend the TestGrid
-configuration file at https://github.com/kubernetes/test-infra/blob/master/testgrid/config/config.yaml and a list of GCS buckets at https://github.com/kubernetes/test-infra/blob/master/buckets.yaml.
+configuration file at https://github.com/kubernetes/test-infra/blob/master/testgrid/config/config.yaml and a list of GCS buckets at https://github.com/kubernetes/test-infra/blob/master/kettle/buckets.yaml.
```
Comment (BenTheElder) [inline docs/tests/node-e2e-ci.md]: why is testgrid config reading from kettle/buckets? aren't these nominally testgrid or gubernator buckets? /shrug

Comment (rmmh) [reply docs/tests/node-e2e-ci.md]: Testgrid isn't reading from kettle, but kettle needs its own list of buckets to read.

Comment (BenTheElder) [reply docs/tests/node-e2e-ci.md]: Yeah OK. For some reason I thought the testgrid config validation read this.

File: kettle/Makefile [hunk 2]
```diff
@@ -36,7 +34,14 @@ push: build
 	echo Pushed $(IMG):latest and $(IMG):$(TAG)
 
 deploy: get-cluster-credentials
-	sed "s/:latest/:$(TAG)/g" deployment.yaml | kubectl apply -f -
+	sed "s/:latest/:$(TAG)/g" deployment.yaml | kubectl apply -f - --record
+
+update: get-cluster-credentials
+	kubectl set image deployment/kettle kettle=$(IMG):$(TAG)
```
Comment (BenTheElder) [inline kettle/Makefile]: @paulangton this is what you wanted earlier effectively

File: docs/tests/node-e2e-ci.md (hunk 1 above)
Comment (fejta) [reply docs/tests/node-e2e-ci.md]: > kettle needs its own list of buckets to read

Couldn't it just discover this by reading `gs://k8s-testgrid/config` and/or parsing `testgrid/config.yaml`?

Comment (rmmh) [reply docs/tests/node-e2e-ci.md]: Testgrid doesn't care about buckets, and doesn't have prefixes. Kettle doesn't care about almost all of the Testgrid config, and would have to use heuristics to move from job directories to buckets to scan.

**Comment by rmmh** [comment]
Tiny bug fix, self-LGTMing.

File: docs/tests/node-e2e-ci.md (hunk 1 above)
Comment (fejta) [reply docs/tests/node-e2e-ci.md]: Shouldn't it care about all of the testgrid config though?

Comment (rmmh) [reply docs/tests/node-e2e-ci.md]: Clarified this README about how testgrid doesn't require /kettle/buckets.yaml at all.

**Pull request merged**